### PR TITLE
ICU-22741 Update CLDR-ICU instructions to mention some hardcoded lists that may need updating

### PR DESCRIPTION
**Pull request by pedberg-icu**
##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22741
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [x] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [x] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Review by macchiati — COMMENTED** [review]
Looks good. Are these also in ICU4J?

**Comment by pedberg-icu** [comment]
> Looks good. Are these also in ICU4J?

ICU4J just has the measure unit lists (including currency) that get generated as part of the process in https://unicode-org.github.io/icu/processes/release/tasks/updating-measure-unit.html

**Comment by macchiati** [comment]
great, thanks

**Review by srl295 — DISMISSED** [review]

File: docs/processes/cldr-icu.md [hunk 1]
```diff
@@ -285,6 +285,24 @@ Paste the following block after the dictionaries block and before the final clos
     }
 ```
 
+5f. Update hard-coded lists in ICU
+
+ICU4 has some hard-coded lists of locale-related codes that may need updating. Ideally these should
+be replaced by data converted from CLDR (and there is at least one ticket for doing that). In the
```
Comment (srl295) [inline docs/processes/cldr-icu.md:291]:
```suggestion
be replaced by data converted from CLDR ([ICU-22839](https://unicode-org.atlassian.net/browse/ICU-22839)). In the
```

**Comment by srl295** [comment]
Suggested to link to an actual ticket.

**Comment by pedberg-icu** [comment]
> Suggested to link to an actual ticket.

@srl295 Thanks, good idea, added that. Now needs re-stamp...🙂

**Review by srl295 — APPROVED** [review]

**Pull request merged**